chore(docs) IPv4 clarity (#1877)
questdb/questdb.io: docs/reference/operators/ipv4.md

This document outlines the IPv4 data type operators.

IPv4 addresses are presented between the range `0.0.0.1 - 255.255.255.255`.

When passed, they may contain trailing and leading dots: `...1.1.1.1...`.

> IPv4 addresses are stored as `int`, but can be passed as `string`. When passed as `string`, they may contain a subnet mask (`35.24.65.2/16`) or a subnet with a netmask(`2.232/16`). See below to see which operators support `string` type arguments and which do not.

Reviewer: ```suggestion
IPv4 addresses are stored as `int` but are passed as `string` in dotted decimal notation.
```

Reviewer: I think it might be better to only mention netmask stuff with the specific operators/functions that support it so as to avoid confusion

Reviewer: I can't add comments on those lines as they were not changed in this pr but would it be possible to add examples of string arguments under <<. <<=, rnd_ipv4(string, int) and netmask() as those are the only operators/functions that support string args? 

Also, just noticed that we don't have netmask() on this page - would it be possible to add it please? See this pr for detail https://github.com/questdb/questdb/pull/3624

Reviewer: And also maybe an example for rnd_ipv4() as it's the only one that doesn't have one? Thanks!

Author: If we don't have anything about subnets/netmasks/strings at the top of this page, we don't discuss them until they appear in the examples. I'd say it's okay to intro them, especially because they do appear in the function examples too:

```
SELECT ip, bytes, CASE WHEN ip << '2.65.32.1/2' THEN 1 ELSE 0 END FROM example
```

I'll defer to you if you feel strongly enough, but IMO this is fair balance.

---

Will also add the additions. 👍

Author: @allegraharris Okay -- that's probably better.

WDYT?

Reviewer: Great! Thank you

This document outlines the IPv4 data type operators.

> IPv4 addresses are presented between the range `0.0.0.1 - 255.255.255.255`.

Reviewer 2: I would be explicit about the address `0.0.0.0` being interpreted as `null`. 
_What a strange range! Why not 0.0.0.0?_ <-- This was the first thing on my mind whilst reading this.

Author: Good call! Adding.

> When passed, they may contain trailing and leading dots: `...1.1.1.1...`.

Reviewer 2: do we really want to document this? what is it good for? maybe it could stay as an _undocumented implementation detail_ so we have the freedom to change it if this is not desirable for any reason?

Author: I can't think of a practical use case other.

If we were voted I'd vote "not document".

Reviewer: This was implemented because browsers are tolerant of leading and trailing dots. I'm happy to keep/remove from docs

Author: Huh - TIL! 

Okay, pushed up.

> ipv4 '44.8.9.10' != ipv4 '6.2.90.1' -> T

Reviewer 2: not directly related to this PR, but perhaps worth fixing too. 
this section is non-obvious. for 2 reasons:
## the first reason
`ipv4 '44.8.9.10' != ipv4 '6.2.90.1' -> T`
What is `T`? I figured out it's `true`, but it took me some effort to realize it. That's not good. 

Compare IPv4 example:
<img width="508" alt="image" src="https://github.com/questdb/questdb.io/assets/158619/b3d54d36-f0af-4999-ade4-8d551ec7093a">

With an existing example at https://deploy-preview-1877--unruffled-blackwell-31bfb2.netlify.app/docs/reference/operators/comparison/ 
<img width="610" alt="image" src="https://github.com/questdb/questdb.io/assets/158619/aa648f92-46cf-4800-b9ca-a0aa3b8864b4">

Differences: The 2nd example does not force me to think. It's very obvious. Also, it gives me a proper SQL I can actually execute. Instead of an abstract syntax. I think that makes it better documentation. 

## The 2nd reason
How is IP address comparison defined? I am too ignorant to be aware of any IP comparison standard. And my intuition is failing me: `127.0.0.1 > 127.0.1.0` is this `true` or `false`? I am not sure.

Author: 👍 Saved these notes for new PR, as broad pass is needed to address many quality-of-life changes.

Thanks!

```sql
ipv4 '92.11.8.40' - ipv4 '92.11.8.0' -> 40
```

> ### Return netmask - netmask()

Reviewer: ```suggestion
### Return netmask - netmask(string)
```

> Takes a `string` argument in the format `22.59.138.9/8`.

Reviewer: I think we should note that we also accept subnet + netmask e.g., `2.2/16'
